modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C

```diff
@@ -161,11 +166,17 @@ PeripheralRingMeshGenerator::generate()
   if (*(input_mesh->elem_dimensions().begin()) != 2 ||
       *(input_mesh->elem_dimensions().rbegin()) != 2)
     paramError("input", "Only 2D meshes are supported.");
+  // Move the centroid of the mesh to (0, 0, 0) if input centroid is enforced to be the ring center.
+  const Point input_mesh_centroid = MooseMeshUtils::meshCentroidCalculator(*input_mesh);
+  if (_force_input_centroid_as_center)
+    MeshTools::Modification::translate(
+        *input_mesh, -input_mesh_centroid(0), -input_mesh_centroid(1), -input_mesh_centroid(2));

```

Comment:
dont you need to retranslate back after too?

Reply:
I don't think we need to translate it back.

Reply:
so the peripheral mesh is created around 0,0,0 always, it s just a matter of whether the mesh should be centered there too?

Reply:
Yes. It is just a matter whether to center the input mesh if it is not originally centered.